bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/onvif/OnvifConnection.java

```diff
@@ -312,10 +316,13 @@ public void processReply(String message) {
         } else if (message.contains("GetSystemDateAndTimeResponse")) {// 1st to be sent.
             connecting.lock();
             try {
+                connectError = false;
+                refusedError = false;
                 isConnected = true;
             } finally {
                 connecting.unlock();
             }
```

Comment:
```suggestion
setIsConnected(true);
```
Also need to add the following to the new setIsConnected(boolen) function:

```
connectError = false;
refusedError = false;
```

Reply:
I change both locations in the processReply() function to use the new function setIsConnected().
I reset connectError and refusedError also for case 'new connection state == false' as this leads to new reconnection attempts.